packages/runtimes/hmr/src/loaders/hmr-runtime.js

```diff
@@ -27,8 +27,10 @@ var checkedAssets, assetsToAccept, acceptedAssets;
 // eslint-disable-next-line no-redeclare
 var parent = module.bundle.parent;
 if ((!parent || !parent.isParcelRequire) && typeof WebSocket !== 'undefined') {
-  var hostname = location.hostname;
-  var port = location.port ? ':' + location.port : '';
+  var hostname = location.origin.startsWith('http')
+    ? location.hostname
+    : 'localhost';
+  var port = location.port ? ':' + location.port : ':1234';
```

Comment:
We'll need to get this value from the options rather than hard coding 1234. I'd recommend writing a variable like `__PARCEL_HMR_PORT` [here](https://github.com/parcel-bundler/parcel/blob/v2/packages/runtimes/hmr/src/HMRRuntime.js#L23) and accessing it in place of this string. You can get the port from `options.hot.port`.

Reply:
So options.hot.port is in scope for the client at this point?

Reply:
No:
https://github.com/parcel-bundler/parcel/blob/003305315a3f96db9a3d5e40488c8e9b28d7dfd7/packages/runtimes/hmr/src/loaders/hmr-runtime.js#L28-L32
https://github.com/parcel-bundler/parcel/blob/003305315a3f96db9a3d5e40488c8e9b28d7dfd7/packages/core/parcel/src/cli.js#L283-L284
But we shouldn't set `process.env.HMR_HOSTNAME ` in the cli but rather in `@parcel/core` or an existing plugin.

Reply:
I couldn't see how to get environment properties into the client but I found I could add the variables here: https://github.com/parcel-bundler/parcel/blob/e3cbe12600c1ea30c42a22c822bdfe9016761dab/packages/runtimes/hmr/src/HMRRuntime.js#L23